src/Core/Content/Media/Api/MediaVideoCoverController.php

```diff
@@ -0,0 +1,42 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Media\Api;
+
+use Shopware\Core\Content\Media\Service\VideoCoverService;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+#[Route(defaults: ['_routeScope' => ['api']])]
+#[Package('discovery')]
+class MediaVideoCoverController extends AbstractController
+{
+    /**
+     * @internal
+     */
+    public function __construct(private readonly VideoCoverService $videoCoverService)
+    {
+    }
+
+    #[Route(path: '/api/_action/media/{mediaId}/video-cover', name: 'api.action.media.set_video_cover', methods: ['POST'], defaults: ['_acl' => ['media.editor']])]
+    public function assignVideoCover(string $mediaId, Request $request, Context $context): JsonResponse
+    {
+        $payload = $request->request->all();
+        $coverMediaId = $payload['coverMediaId'] ?? null;
```

Comment:
You can just use `$coverMediaId = $request->get('coverMediaId');`. This will return the coverMediaId directly and has null as fallback. Then you do not need the payload nor the condition.

Reply:
but then please use `$request->request->get('coverMediaId');`

the `get` method directly from the request class is deprecated with Symfony 7.4